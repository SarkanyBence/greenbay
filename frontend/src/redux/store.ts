import { configureStore } from "@reduxjs/toolkit";
import allReducers from "./combineReducers";

const store = configureStore({
  reducer: allReducers,
});

store.subscribe(() => {
  console.log("state updated", store.getState());
});

export default store;

export type StateType = ReturnType<typeof store.getState>;
export type DispatchType = typeof store.dispatch;
