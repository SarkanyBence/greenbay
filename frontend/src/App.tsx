import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect,
} from "react-router-dom";
import Buy from "./components/Buy";
import Header from "./components/Header";
import Login from "./components/Login";
import Register from "./components/Register";
import Sell from "./components/Sell";
import { useAppSelector } from "./hooks/stateHooks";
import { StateType } from "./redux/store";

function App() {
  const isLoggedIn: boolean = useAppSelector(
    (state: StateType) => state.user.isLoggedIn
  );

  if (!isLoggedIn) {
    return (
      <Router>
        <div className="App">
          <p>IS LOGGED IN: {isLoggedIn}</p>
          <Header />
          <Switch>
            <Route path="/register">
              <Register />
            </Route>
            <Route path="/login">
              <Login />
            </Route>
            <Route path="/*">
              <Redirect to="/login" />
            </Route>
          </Switch>
        </div>
      </Router>
    );
  } else {
    return (
      <Router>
        <div className="App">
          <p>IS LOGGED IN: {isLoggedIn}</p>
          <Header />
          <Switch>
            <Route path="/buy">
              <Buy />
            </Route>
            <Route path="/sell">
              <Sell />
            </Route>
          </Switch>
        </div>
      </Router>
    );
  }
}

export default App;
