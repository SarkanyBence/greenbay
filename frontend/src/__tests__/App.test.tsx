import renderer from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";
import { render, screen } from "@testing-library/react";
import App from "../App";
import { Provider } from "react-redux";
import store from "../redux/store";

describe("App component tests", () => {
  it("App snapshot test", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <App />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it.skip ("App renders correctly test", () => {
    render(<App />);
    const elements = screen.getAllByText("greenBay");
    expect(elements).toHaveTextContent("greenBay");
  });
});
