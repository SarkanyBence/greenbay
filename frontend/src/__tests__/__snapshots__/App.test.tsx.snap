// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component tests App snapshot test 1`] = `
<div
  className="App"
>
  <div>
    loading
  </div>
</div>
`;
